R/checks.R

```diff
@@ -1,101 +1,107 @@
-check_numeric <- function(x, input = "`x`") {
+check_numeric <- function(x, input = "`x`", call) {
   if (!is.vector(x) || !is.numeric(x)) {
-    rlang::abort(paste0(input, " should be a numeric vector."))
+    cli::cli_abort("{.arg {input}} should be {an} numeric vector.")
   }
   invisible(NULL)
 }
```

Comment:
this function could be replaced with `check_number_decimal()` from the standalone checker

Reply:
`check_number_decimal()` requires a single value; I'm checking for a numeric vector: 

``` r
library(rlang)

desirability2:::check_number_decimal(runif(2))
#> Error in `call_match()`:
#> ! `fn` must be supplied.
desirability2:::check_numeric(runif(2))
```

<sup>Created on 2024-10-19 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

It's in a helper so that when/if the standalone file/package support vectors, it will be easy to replace